### PR TITLE
Fix dashboard items

### LINKED ISSUE #1003: Dashboard Items Not Clickable Despite Sidebar Navigation Functionality
**Issue Description:**
There are several items shown in the dashboard like members admins posts events etc. These items are not clickable and do not redirects to respective pages like the sidebar.

**Issues Identified:**
Non-Clickable Dashboard Items:
The dashboard items (e.g., "Posts," "Members," "Admins") are not responsive to clicks, rendering them non-functional. Users are unable to access the associated content directly from the dashboard.

**Expected Behavior:**
Clicking on a dashboard item should lead to the relevant section or content within the dashboard.

**Steps to Reproduce:**
Navigate to the dashboard.
Observe that the dashboard items (e.g., "Posts," "Members," "Admins") are not clickable.
Attempt to click on any dashboard item to access its corresponding content.

image

**The dashboard items should be functional like the sidebar items.**

### PR DESCRIPTION
**Pull request by skbhagat0502**
**What kind of change does this PR introduce?**
In this PR ,  I have made the dashboard button such as "Members", "Post" etc clickable and now it redirects to the respective page. Like clicking on the post button redirects to the posts page. 
image
Here I have added a video for reference.

https://github.com/PalisadoesFoundation/talawa-admin/assets/109683163/2c381774-0a18-4737-b8a9-9023c4f76a95

Previously the dashboard buttons shown in image were not working. Now all the buttons are working well. 
**Issue Number:**

Fixes #1003 

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by rishav-jha-mech** [comment]
@skbhagat0502 please upload the images and videos of the changes that you have made.

**Comment by beingnoble03** [comment]
@skbhagat0502 please update the PR description.
"What kind of change does this PR introduce?"
It should clearly explain what changes the PR brings. This is very important for other contributors.

**Comment by skbhagat0502** [comment]
@rishav-jha-mech @beingnoble03 I have edited the PR description as well as added a video and image of the changes made.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 1]
```diff
@@ -120,42 +122,90 @@ function organizationDashboard(): JSX.Element {
...
             ) : (
               <Row style={{ display: 'flex' }}>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`/orgpeople/id=${organizationId}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:131]: Can you create variable for the url/link e.g. 
```ts
const peopleLink: string = `/orgpeople/id=${organizationId}`;
```

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 2]
```diff
@@ -120,42 +122,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`/orgevents/id=${organizationId}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:176]: Similarly create a variable here too.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 3]
```diff
@@ -120,42 +122,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`/orgpost/id=${organizationId}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:161]: Similarly create a variable for this too.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 4]
```diff
@@ -120,42 +122,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`/orgpeople/id=${organizationId}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:146]: Similarly create a variable for this too.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 5]
```diff
@@ -120,42 +122,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`/blockuser/id=${organizationId}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:191]: Similarly create a variable for this too.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 6]
```diff
@@ -120,42 +122,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push('/requests');
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:206]: Similarly create a variable for this too.

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
Creating different variables here will enhance the code quality and reduce the code repetition.

**Comment by skbhagat0502** [comment]
@beingnoble03 I added variables for the all the links as you said.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx (hunk 6 above)
Comment (skbhagat0502) [reply src/screens/OrganizationDashboard/OrganizationDashboard.tsx]: I added the variable for this.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx (hunk 5 above)
Comment (skbhagat0502) [reply src/screens/OrganizationDashboard/OrganizationDashboard.tsx]: I added the variable for this.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx (hunk 4 above)
Comment (skbhagat0502) [reply src/screens/OrganizationDashboard/OrganizationDashboard.tsx]: I added the variable for this.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx (hunk 3 above)
Comment (skbhagat0502) [reply src/screens/OrganizationDashboard/OrganizationDashboard.tsx]: I added the variable for this.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx (hunk 2 above)
Comment (skbhagat0502) [reply src/screens/OrganizationDashboard/OrganizationDashboard.tsx]: I added the variable for this.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx (hunk 1 above)
Comment (skbhagat0502) [reply src/screens/OrganizationDashboard/OrganizationDashboard.tsx]: I added the variable for this.

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 7]
```diff
@@ -120,42 +128,90 @@ function organizationDashboard(): JSX.Element {
...
             ) : (
               <Row style={{ display: 'flex' }}>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`${peopleLink}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:137]:
```suggestion
                    history.push(peopleLink);
```

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 8]
```diff
@@ -120,42 +128,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`${peopleLink}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:152]:
```suggestion
                    history.push(peopleLink);
```

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 9]
```diff
@@ -120,42 +128,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`${postsLink}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:167]:
```suggestion
                    history.push(postsLink);
```

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 10]
```diff
@@ -120,42 +128,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`${eventsLink}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:182]: Similar change

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 11]
```diff
@@ -120,42 +128,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`${blockUserLink}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:197]: Similar change as above

File: src/screens/OrganizationDashboard/OrganizationDashboard.tsx [hunk 12]
```diff
@@ -120,42 +128,90 @@ function organizationDashboard(): JSX.Element {
...
                   />
                 </Col>
-                <Col xs={6} sm={4} className="mb-4">
+                <Col
+                  xs={6}
+                  sm={4}
+                  role="button"
+                  className="mb-4"
+                  onClick={(): void => {
+                    history.push(`${requestLink}`);
```
Comment (beingnoble03) [inline src/screens/OrganizationDashboard/OrganizationDashboard.tsx:212]: Similar change as above

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
Accept the suggested changes. Rest LGTM.

**Comment by skbhagat0502** [comment]
@beingnoble03 I made the changes as said by you.

**Review by beingnoble03 — APPROVED** [review]
Approving. But please try to follow commit message convention. You can refer [this](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/) for writing a good commit message.

**Comment by beingnoble03** [comment]
Please fix the failing tests.

**Comment by palisadoes** [comment]
Merging will need to wait until this PR is merged.  @JamarTG has already had to fix PR conflicts before, and his feature is a good one. @skbhagat0502 you may have to refactor afterwards.

- https://github.com/PalisadoesFoundation/talawa-admin/pull/979

**Comment by skbhagat0502** [comment]
> Approving. But please try to follow commit message convention. You can refer [this](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/) for writing a good commit message.

Thanks for the reference for writting good commit messages. I will surely follow the convention.

**Comment by skbhagat0502** [comment]
ok @palisadoes , @beingnoble03

**Comment by skbhagat0502** [comment]
> Please fix the failing tests. I guess no-inferrable-types check is on. You can remove `: string` in the variable assignment you did earlier.

I removed `: string` from variable assignment.

**Review by rishav-jha-mech — APPROVED** [review]

**Comment by skbhagat0502** [comment]
> > Please fix the failing tests. I guess no-inferrable-types check is on. You can remove `: string` in the variable assignment you did earlier.
> 
> I removed `: string` from variable assignment.

Did I made any mistake, as It says I dismissed @beingnoble03 stale review.

**Comment by beingnoble03** [comment]
> > > Please fix the failing tests. I guess no-inferrable-types check is on. You can remove `: string` in the variable assignment you did earlier.
> > 
> > 
> > I removed `: string` from variable assignment.
> 
> Did I made any mistake, as It says I dismissed @beingnoble03 stale review.

No, you didn't do anything wrong. Whenever you push a new commit, the previously approved review is dismissed.

**Review by beingnoble03 — APPROVED** [review]

**Comment by rishav-jha-mech** [comment]
@skbhagat0502 fix the failing tests.

**Comment by skbhagat0502** [comment]
@rishav-jha-mech I run the test locally and it is not failing. So I am not able to fix it. If I will fix the failing test that is shown I would have used javascript chaining but I think I should think before doing this as that part of code was already present. I have not changed it in this PR. What's your opinion should I write like this? 
`const currentUrl = window.location.href?.split('=')[1];`
As the error is comming from this part. How can I check it because locally I found that it is not failing.

**Comment by palisadoes** [comment]
@skbhagat0502 We won't be able to merge this until the tests are fixed. The error messages seem self explanatory, there is a repetitive condition that is failing.

**Comment by skbhagat0502** [comment]
Ok @palisadoes I will fix it soon.

**Comment by skbhagat0502** [comment]
@palisadoes I can't see the test falling in my pc when I run `npm run test` locally. That's why I am not able to resolve it. Please help me in this.

**Comment by palisadoes** [comment]
Please work on getting the code test coverage on the files you edited up to 95%. You have only 36% at the moment.

image

**Comment by skbhagat0502** [comment]
Ok @palisadoes

**Comment by skbhagat0502** [comment]
@palisadoes there are no lines to cover for test now in `OrganisationDashboard.tsx` component. The code coverage percentage is less because of other components. Please tell me if there is something else to be done.

**Comment by palisadoes** [comment]
@rishav-jha-mech @beingnoble03 Please take another look and verify the testing methodology

**Comment by skbhagat0502** [comment]
@rishav-jha-mech @beingnoble03 Please review.

**Review by beingnoble03 — APPROVED** [review]
LGTM. The test is failing due to low level of code coverage, but the files which are not covered, are not part of this PR. This can be merged.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**